### ISSUE TITLE
Ignore unrelated folder in target

### ISSUE THREAD

**Issue by dpprdan** [robweber/xbmcbackup#244]
I had the following error, because I had a folder called `Orig` in my target folder.

```
2024-10-11 08:36:52.203 T:17774   error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'ValueError'>
                                                   Error Contents: invalid literal for int() with base 10: 'Orig'
                                                   Traceback (most recent call last):
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/script.xbmcbackup/default.py", line 139, in <module>
                                                       backup.backup()
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/script.xbmcbackup/resources/lib/backup.py", line 208, in backup
                                                       self._rotateBackups()
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/script.xbmcbackup/resources/lib/backup.py", line 487, in _rotateBackups
                                                       dirs = self.listBackups(reverse=False)
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/script.xbmcbackup/resources/lib/backup.py", line 88, in listBackups
                                                       folderName = self._dateFormat(aDir)
                                                                    ^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/script.xbmcbackup/resources/lib/backup.py", line 472, in _dateFormat
                                                       date_time = datetime(int(dirName[0:4]), int(dirName[4:6]), int(dirName[6:8]), int(dirName[8:10]), int(dirName[10:12]))
                                                                            ^^^^^^^^^^^^^^^^^
                                                   ValueError: invalid literal for int() with base 10: 'Orig'
                                                   -->End of Python script error report<--
```

Would it make sense to add something like

```
len(folderName) >= 12 and folderName[0:12].isdigit()
```

here (probably?) 

https://github.com/robweber/xbmcbackup/blob/1559df4d4c72547424a5d2b8dbfe7fc09c2d17c5/resources/lib/backup.py#L85

as is the case for the zip files,

https://github.com/robweber/xbmcbackup/blob/1559df4d4c72547424a5d2b8dbfe7fc09c2d17c5/resources/lib/backup.py#L96

so that unrelated folders within the target folder do not cause an error? E.g. the error comes up, even if one chooses to backup into zip files instead of plain folders.

**Comment by robweber** [comment]
Looking through the workflow on this I'm a bit confused how that part of the `_listBackups()` function got executed. The first check on a directory is if an `xbmcbackup.val` function exists within the directory. If the validation file exists the contents of the directory _should_ contain a valid backup. Does your `Orig` file contain a validation file?

**Comment by dpprdan** [comment]
I cannot say, because I've since deleted the `Orig` folder. It was probably a very old backup and it's therefore quite possible that it contained a `xbmcbackup.val` file. It's also possible that I changed the folder name manually years ago.

Given that I cannot really reproduce this and that 

https://github.com/robweber/xbmcbackup/blob/1559df4d4c72547424a5d2b8dbfe7fc09c2d17c5/resources/lib/backup.py#L85

does in spirit what wanted, I guess we can file this under "very unlikely edge case" and close this. Thanks for looking into this!
